modules/mtl/arrow-mtl-data/src/test/kotlin/arrow/mtl/WriterTTest.kt

```diff
@@ -41,63 +40,71 @@ import arrow.test.laws.MonadFilterLaws
+        optionEQK()
       ),
       DivisibleLaws.laws(
         WriterT.divisible<ConstPartialOf<Int>, Int>(Const.divisible(Int.monoid())),
-        { WriterT(it.const()) },
-        Eq { a, b -> a.value().value() == b.value().value() }
+        WriterT.genK(Const.genK(Gen.int()), Gen.int()),
+        constEQK()
       ),
-      ConcurrentLaws.laws(WriterT.concurrent(IO.concurrent(), Int.monoid()), IOEQ(), IOEQ(), IOEQ()),
+      ConcurrentLaws.laws(WriterT.concurrent(IO.concurrent(), Int.monoid()), ioEQK()),
       MonoidKLaws.laws(
         WriterT.monoidK<ForListK, Int>(ListK.monoidK()),
-        WriterT.applicative(ListK.monad(), Int.monoid()),
-        Eq { a, b -> a.value() == b.value() }),
+        WriterT.genK(ListK.genK(), Gen.int()),
+        listEQK()
+      ),
 
       MonadWriterLaws.laws(WriterT.monad(Option.monad(), Int.monoid()),
         WriterT.monadWriter(Option.monad(), Int.monoid()),
         Int.monoid(),
         Gen.intSmall(),
         Gen.tuple2(Gen.intSmall(), Gen.intSmall()),
-        Eq { a, b ->
-          a.value().fix().let { optionA: Option<Tuple2<Int, Int>> ->
-            val optionB = b.value().fix()
-            optionA.fold({ optionB.fold({ true }, { false }) }, { value: Tuple2<Int, Int> -> optionB.fold({ false }, { value == it }) })
-          }
-        },
-        Eq { a, b ->
-          a.value().fix().let { optionA: Option<Tuple2<Int, Tuple2<Int, Int>>> ->
-            val optionB = b.value().fix()
-            optionA.fold({ optionB.fold({ true }, { false }) }, { value: Tuple2<Int, Tuple2<Int, Int>> -> optionB.fold({ false }, { value == it }) })
-          }
-        }
+        optionEQK(),
+        Int.eq()
       ),
 
       MonadFilterLaws.laws(
         WriterT.monadFilter(Option.monadFilter(), Int.monoid()),
         { WriterT(Option(Tuple2(it, it))) },
-        object : Eq<Kind<WriterTPartialOf<ForOption, Int>, Int>> {
-          override fun Kind<WriterTPartialOf<ForOption, Int>, Int>.eqv(b: Kind<WriterTPartialOf<ForOption, Int>, Int>): Boolean =
-            value().fix().let { optionA: Option<Tuple2<Int, Int>> ->
-              val optionB = b.value().fix()
-              optionA.fold({ optionB.fold({ true }, { false }) }, { value: Tuple2<Int, Int> -> optionB.fold({ false }, { value == it }) })
-            }
-        })
+        optionEQK()
+      )
     )
   }
 }
+
+private fun <F, W> WriterT.Companion.genK(
+  GENKF: GenK<F>,
+  GENW: Gen<W>
+) = object : GenK<WriterTPartialOf<F, W>> {
+  override fun <A> genK(gen: Gen<A>): Gen<Kind<WriterTPartialOf<F, W>, A>> =
+    GENKF.genK(Gen.tuple2(GENW, gen)).map(::WriterT)
+}
+
+fun <F, W> WriterT.Companion.eqK(EQKF: EqK<F>, EQW: Eq<W>) = object : EqK<WriterTPartialOf<F, W>> {
+  override fun <A> Kind<WriterTPartialOf<F, W>, A>.eqK(other: Kind<WriterTPartialOf<F, W>, A>, EQ: Eq<A>): Boolean =
+    (this.fix() to other.fix()).let {
+      EQKF.liftEq(Tuple2.eq(EQW, EQ)).run {
+        it.first.value().eqv(it.second.value())
+      }
+    }
+}
```

Comment:
Same as for eitherT, if writerT has an eq instance somewhere move this there as well, leave it if not

Reply:
ok